bin/vagrant-provision.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+set -e
+
+# Map binary name to source URL.
+REPOS=(
+	"docker-compose,https://github.com/docker/compose/releases/download/1.21.2/docker-compose-`uname -s`-`uname -m`"
+	"composer,https://getcomposer.org/download/1.6.5/composer.phar"
+)
+
+# Add Docker repo.
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+
+# Add Glide for Golang
+sudo add-apt-repository ppa:masterminds/glide -y
+
+sudo apt-get update
+sudo apt-get install -y \
+	docker-ce avahi-daemon git php-cli php-xml zip golang-go glide
+
+# Install various binary scripts.
+for REPOSET in "${REPOS[@]}"; do
+	while IFS=',' read -ra REPO; do
+		BINPATH="/usr/local/bin/${REPO[0]}"
+
+		if [ ! -f "$BINPATH" ]; then
+			echo "Installing ${REPO[1]} to $BINPATH"
+
+			sudo curl -sL "${REPO[1]}" -o "$BINPATH"
+			sudo chmod +x "$BINPATH"
+		fi
+	done <<< "$REPOSET"
+done
+
+# Add user to docker group to avoid using sudo.
+sudo usermod -aG docker $USER
+newgrp docker
+
+# Setup Golang environment.
+mkdir -p ~/go/src
+export GOPATH=~/go
+echo "export GOPATH=~/go" >> ~/.bashrc
+echo "export PATH=\$PATH:\$GOPATH/bin" >> ~/.bashrc
+
+# Ensure we have the environment defined.
+if [ ! -f /vagrant/.env ]; then
+	cp /vagrant/.env.dist /vagrant/.env
+fi
+
+# Move to the root of the project.
+cd /vagrant
+
+# Build all service Docker images and Go binaries.
+make build.images
```

Comment:
This is currently failing with the following error:

```bash
$ make build.images
Installing dependencies ...
[WARN]	The name listed in the config file (github.com/xwp/go-tide) does not match the current location (.)
[INFO]	Downloading dependencies. Please wait...
[INFO]	--> Found desired version locally cloud.google.com/go 0fd7230b2a7505833d5f69b75cbd6c9582401479!
...
[INFO]	--> Setting version for gopkg.in/toqueteos/substring.v1 to c5f61671513240ddf5563635cc4a90e9f3ae4710.
[INFO]	--> Setting version for cloud.google.com/go to 0fd7230b2a7505833d5f69b75cbd6c9582401479.
[INFO]	Exporting resolved dependencies...
[INFO]	--> Exporting github.com/aws/aws-sdk-go
[INFO]	--> Exporting github.com/toqueteos/trie
...
[INFO]	Replacing existing vendor dependencies
Cleaning up Lighthouse Server GO binary 'bin/lighthouse-server' ...
Cleaning up PHPCS Server GO binary 'bin/phpcs-server' ...
Cleaning up Sync Server GO binary 'bin/sync-server' ...
Building Lighthouse Server GO binary 'bin/lighthouse-server' ...
cmd/lighthouse-server/main.go:19:2: cannot find package "context" in any of:
	/usr/lib/go-1.6/src/context (from $GOROOT)
	/home/vagrant/go/src/context (from $GOPATH)
cmd/lighthouse-server/main.go:10:2: cannot find package "github.com/wptide/pkg/env" in any of:
	/usr/lib/go-1.6/src/github.com/wptide/pkg/env (from $GOROOT)
	/home/vagrant/go/src/github.com/wptide/pkg/env (from $GOPATH)
...
service/lighthouse-server/Makefile:29: recipe for target 'lighthouse.build.bin' failed
make: *** [lighthouse.build.bin] Error 1
```

@rheinardkorf Do you have ideas on why that is happening? It appears to be looking at `/home/vagrant/go/src` and `/usr/lib/go-1.6/src` for dependancies while they're actually stored at `/vagrant/vendor`.